src/prefect/client/client.py

```diff
@@ -330,6 +330,18 @@ def _send_request(
             )
 
         # Check if request returned a successful status
+        if response.status_code == 400:
+            try:
+                msg = (
+                    f"400 Client Error: Bad Request for url: {url}\n\n"
+                    f"This is likely caused by a poorly formatted GraphQL query or mutation. "
+                    f"GraphQL sent:\n\n{parse_graphql(params)}"
+                )
+                raise ClientError(msg)
```

Comment:
Any consideration to attach additional metadata directly to ClientErrors then just have a robust string formatter within the class?

Reply:
Not sure I see any benefit to that